js/packages/openinference-instrumentation-langchain/src/instrumentationUtils.ts

```diff
@@ -0,0 +1,47 @@
+import type * as CallbackManagerModuleV2 from "@langchain/core/callbacks/manager";
+import type * as CallbackManagerModuleV1 from "@langchain/coreV1/callbacks/manager";
+import { Tracer } from "@opentelemetry/api";
+import { LangChainTracer } from "./tracer";
+
+export function addTracerToHandlers(
+  tracer: Tracer,
+  handlers?: CallbackManagerModuleV1.Callbacks,
+): CallbackManagerModuleV1.Callbacks;
+export function addTracerToHandlers(
+  tracer: Tracer,
+  handlers?: CallbackManagerModuleV2.Callbacks,
+): CallbackManagerModuleV2.Callbacks;
+export function addTracerToHandlers(
+  tracer: Tracer,
+  handlers?:
+    | CallbackManagerModuleV1.Callbacks
+    | CallbackManagerModuleV2.Callbacks,
+): CallbackManagerModuleV1.Callbacks | CallbackManagerModuleV2.Callbacks {
+  if (handlers == null) {
+    // There are some slight differences in the Callbacks interface between v1 and v2
+    return [new LangChainTracer(tracer)];
+  }
+  if (Array.isArray(handlers)) {
+    const tracerAlreadyRegistered = handlers.some(
+      (handler) => handler instanceof LangChainTracer,
+    );
+    if (!tracerAlreadyRegistered) {
+      // There are some slight differences in the CallbackHandler interface between v1 and v2
+      // We support both versions and our tracer is compatible with either as it will extend the class from the installed version
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      handlers.push(new LangChainTracer(tracer) as any);
```

Comment:
See the comment above (now above the function) - there is a slight difference between the CallbackHandlers in v0.1 and v0.2 our tracer should be compatible with either since it properly extends the BaseTracer (a callback handler) from whatever the installed version is 

langchain basically typescript doesn't know if the pre-existing handlers in this case are v0.1 or v0.2  and thinks the tracer is definitely v0.2 so it can't be added to the Array of handlers - casting as v0.1 creates teh same problem in the opposite direction so just went with any for ease here - tried doing some generic stuff but wasn't able to get it to be happy